digdag-core/src/main/java/io/digdag/core/repository/ProjectControl.java

```diff
@@ -69,6 +71,7 @@ public void insertRevisionArchiveData(int revId, byte[] data)
     {
         List<StoredWorkflowDefinition> list = insertWorkflowDefinitionsWithoutSchedules(revision, defs);
         updateSchedules(revision, list, srm, currentTime);
+        validateSla(defs);
```

Comment:
Just in case, if validation fail, updateSchedules() has been cancelled ? 
If this method is only called in transaction and updateSchedules() has no side effect except Pg, I guess it is OK.

Reply:
Thank you for checking, I think I should move this before inserting actually, so I fixed it.
https://github.com/treasure-data/digdag/pull/1390/commits/a1ea9c6584d5e08776adb5056851eaec415f0dc0